src/core/state.js

```diff
@@ -64,6 +64,8 @@ class State {
 
             // no new fields
             if (!fieldsToBeAdded.length) {
+              matchedParentCanvas.fields = updatedFields;
```

Comment:
Can you provide more insight on this line of code, as I could see reassignment of same variable at next line.